File: client/tests/generate-group-bootstrap-fixture.mjs
// Synthetic identities for protocol vectors only. Never use these keys outside tests.
import * as q from '../dist/index.js';
import {writeFileSync} from 'node:fs';
const hex=x=>Buffer.from(x).toString('hex');
const owner=q.generateIdentity(),member=q.generateIdentity();const invite=q.createInvite(owner,'group');
const original=q.createConversation(invite,q.deriveConversationKeys(invite));
const state=new q.GroupState();state.applyGenesis(q.parseGroupGenesisBody(q.createGroupGenesisBody('Interop company','Synthetic QSP1.2 vector',owner,[member.publicKey])));
const wires=[0,1].map(()=>q.serializeEnvelope(q.createMessage(owner,original,'group_rekey',q.createRekey(owner,original,state).bodyBytes)));
const winner=q.resolveRekeyCandidates(owner,original,state,wires);const at=Math.floor(Date.now()/1000);
const token=q.createCurrentEpochInvite(owner,winner.conversation,state,member.publicKey,winner.envelope.msg_id,at);
const value={purpose:'Synthetic public test keys; never operational',at,owner:Object.fromEntries(Object.entries(owner).map(([k,v])=>[k,hex(v)])),member:Object.fromEntries(Object.entries(member).map(([k,v])=>[k,hex(v)])),token,
 source:{id:hex(original.id),epoch:0,keys:Object.fromEntries(Object.entries(original.keys).map(([k,v])=>[k,hex(v)]))},roster_b64:Buffer.from(q.marshalCanonical(state.snapshot())).toString('base64'),wires:wires.map(w=>Buffer.from(w).toString('base64')),winner_id:hex(winner.envelope.msg_id),root:hex(winner.conversation.keys.root)};
writeFileSync(new URL('./group-bootstrap-typescript.json',import.meta.url),JSON.stringify(value,null,2)+'\n');

File: python-dist/tests/generate_group_bootstrap_fixture.py
"""Generate reverse synthetic vector from the shared TS source context."""
import json,base64
from pathlib import Path
from qntm.group import GroupState,create_rekey
from qntm.group_bootstrap import resolve_rekey_candidates,create_current_epoch_invite
from qntm.message import create_message,serialize_envelope
from qntm.cbor import unmarshal
root=Path(__file__).resolve().parents[2]/'client/tests'
f=json.loads((root/'group-bootstrap-typescript.json').read_text())
owner={k:bytes.fromhex(v) for k,v in f['owner'].items()};member={k:bytes.fromhex(v) for k,v in f['member'].items()}
source={'id':bytes.fromhex(f['source']['id']),'type':'group','currentEpoch':0,'participants':[],'keys':{k:bytes.fromhex(v) for k,v in f['source']['keys'].items()}}
state=GroupState();state.apply_genesis(unmarshal(base64.b64decode(f['roster_b64'])))
wires=[serialize_envelope(create_message(owner,source,'group_rekey',create_rekey(owner,source,state,source['id'])[0])) for _ in range(2)]
winner=resolve_rekey_candidates(owner,source,state,wires)
f['token']=create_current_epoch_invite(owner,winner['conversation'],state,member['publicKey'],winner['envelope']['msg_id'],f['at'])
f['wires']=[base64.b64encode(w).decode() for w in wires];f['winner_id']=winner['envelope']['msg_id'].hex();f['root']=winner['conversation']['keys']['root'].hex()
(root/'group-bootstrap-python.json').write_text(json.dumps(f,indent=2)+'\n')
